apps/ios/SimpleX SE/ShareModel.swift

```diff
@@ -189,13 +201,28 @@ class ShareModel: ObservableObject {
         func isMessage(for item: AChatItem?) -> Bool {
             item.map { $0.chatItem.id == chatItemId } ?? false
         }
+
         CompletionHandler.isEventLoopEnabled = true
         let ch = CompletionHandler()
         if isWithoutFile { await ch.completeFile() }
+        var networkTimeout = CFAbsoluteTimeGetCurrent()
         while await ch.isRunning {
+            if CFAbsoluteTimeGetCurrent() - networkTimeout > 30 {
+                networkTimeout = CFAbsoluteTimeGetCurrent()
+                await MainActor.run {
+                    self.errorAlert = ErrorAlert(
+                        title: "No network connection",
+                        actionsOverride: [
+                            ErrorAlert.Action(titleKey: "Keep Trying", role: .cancel),
+                            ErrorAlert.Action(titleKey: "Dismiss Sheet", role: .destructive) { self.completion() }
```

Comment:
"No network connection" is an existing string.
Perhaps there could be better options for the actions